python-frontend/src/main/java/org/sonar/python/types/TypeShed.java

```diff
@@ -143,7 +143,7 @@ private static void setDeclaredReturnType(Symbol symbol, FunctionDef functionDef
     Set<Symbol> commonSymbols = new HashSet<>();
     symbolsPython3.forEach((localName, python3Symbol) -> {
       Symbol python2Symbol = symbolsPython2.get(localName);
-      if (python2Symbol == null) {
+      if (python2Symbol == null || python2Symbol == python3Symbol) {
```

Comment:
I don't exactly remember, to be honest. At some point in my changes, I actually had the same symbol listed in both Python 2 and Python 3 symbols, which led to creating an ambiguous symbol containing 2 identical definitions. If such case ever happens, I think it makes sense to just keep the unique definition instead.

I basically kept the change as a kind of defensive programming, though it's true I'm not sure it can actually be triggered anymore - nothing breaks in the tests or in the ruling if we remove that condition.

Reply:
ok I agree. However I'd be surprised if we had `python2Symbol` and `python3Symbol` having the same reference